helixbass/tree-sitter-grep
# Exploration: supported languages as "singleton values": macros

In this PR:
- add macros for "safely" generating fixed-size "all supported languages" collections
- a couple ergonomic improvements related to these per comments on #52 

To test:
You should see the `fixed_map!()` macro fail if you try to omit any keys, pass any spurious key/values or not pass corresponding values of the expected type
You should see the `by_supported_language!()` macro fail if you omit any supported languages or pass any unknown keys

Based on `supported-languages-fixed-size`
